activerecord/lib/active_record/autosave_association.rb

```diff
@@ -390,11 +390,8 @@ def save_has_one_association(reflection)
           elsif autosave != false
             key = reflection.options[:primary_key] ? send(reflection.options[:primary_key]) : id
 
-            if (autosave && record.changed_for_autosave?) || new_record? || record_changed?(reflection, record, key)
-              unless reflection.through_reflection
-                record[reflection.foreign_key] = key
-              end
-
+            if (autosave && record.changed_for_autosave?) || (!reflection.through_reflection && record_changed?(reflection, record, key))
```

Comment:
This was a long time ago, but I believe looking at the meaning of `if (autosave && record.changed_for_autosave?) || (!reflection.through_reflection && record_changed?(reflection, record, key))` that this would always be true if `new_record?` was true. Is there an edge case you can see? I can put it back.